### DISCUSSION TITLE
Sending data to a server

### DISCUSSION THREAD

**Discussion by knakajima1225** [jspsych/jsPsych#723] [Q&A]
Hello. I'm using the exact same code from the [website](https://www.jspsych.org/overview/data/#storing-data-permanently-as-a-file) to send data to a remote server using Ajax and PHP. When I looked at the data folder after the experiment, though there is a new csv file, it doesn't have any name (just   .csv) and any content (file size is 0). Could you look into the code and tell me what can be wrong? There's no error message both for js and php.

js file
```js
function saveData(name, data){
  var xhr = new XMLHttpRequest();
  xhr.open('POST', 'write_data.php'); // 'write_data.php' is the path to the php file described above.
  xhr.setRequestHeader('Content-Type', 'application/json');
  xhr.send(JSON.stringify({filename: name, filedata: data}));
}

//inside the on_finish function
saveData(prolificid + random_string, jsPsych.data.get().csv());`
```
php file 
```php
<?php
$post_data = json_decode(file_get_contents('php://input'), true);
  // the directory "data" must be writable by the server
$name = "/var/www/data/".$post_data['filename'].".csv";
$data = $post_data['filedata'];
  // write the file to disk
file_put_contents($name, $data);
?>
```

**Comment by jodeleeuw** [comment]
For debugging purposes, can you try replacing 
```js
xhr.send(JSON.stringify({filename: name, filedata: data}));
``` 
with
```js
xhr.send(JSON.stringify({filename: "test_file", filedata: "test_data"}));
```

I'm thinking that since it is generating a file the PHP must be running to completion. Since it is empty with no file name there must be a problem with the $post_data variable. Perhaps there's something about the jsPsych data or the file name causing an issue.
